### DISCUSSION TITLE
Execute in RAM?

### DISCUSSION THREAD

**Discussion by NollKollTroll** [earlephilhower/arduino-pico#766] [Q&A]
It is possible to tell the cmake system to compile the code to execute from ram instead of XIP from flash. [https://forums.raspberrypi.com/viewtopic.php?t=300974](url)
I can't find where to make the changes to make this happen while building in the Arduino IDE, would love to hear where the file to edit is located, or even better to have this as a checkbox in the Board options menu.

**Comment by earlephilhower** [comment]
Why would you want this, exactly?  You can mark specific functions that need maximum speed to be in-RAM with a function decorator, for example.

I haven't tried it, but in theory if you swap the contents of `lib/memmap_default.ld` with the `.ld` file for in-ram use (from the Pico SDK) it should work.  None of the core would know this, so you'd still freeze both cores while doing flash operations.

**Reply by NollKollTroll** [reply]
I'm using the Pico to drive a 65C02 and act as a complete I/O-block. The code is not very big, but really needs all the performance and repeatability it can get. XIP from flash does not perform quite as fast as from RAM. I've tried using  the __not_in_flash_func decorator on a few functions which makes the code run faster, but it would be nice to be sure that all code runs as fast as possible all the time. All my code except setup is time critical, so I'll try to use a lot of decorators tomorrow :)

Overclocking also allegedly works much better when executing from RAM.

**Reply by earlephilhower** [reply]
FYI, for OCing to the max freq (I get 250MHz on my Pico stock from Rpi, but YMMV) you should also drop the SPI divider from /2 to /4. 

That .LD swap should work.  It will lose OTA support but that is probably not critical for you.

Good luck!

**Reply by NollKollTroll** [reply]
I tried using `memmap_copy_to_ram.ld` from the SDK, but it produced code that would not start the Pico properly. The `memmap_default.ld` from the same folder did work. I tried to understand what was going on in the files but gave up and reverted to the original file.
The function decorator `__not_in_flash_func` seems to work the way I want, so I will use that.
Thanks for the help, and more importantly for all the great work done to arduino-pico!
